packages/core/src/blocks/CodeBlockContent/CodeBlockContent.ts

```diff
@@ -0,0 +1,359 @@
+import { InputRule, isTextSelection } from "@tiptap/core";
+import {
+  BuiltinLanguage,
+  bundledLanguagesInfo,
+  createHighlighter,
+  Highlighter,
+} from "shiki";
+import {
+  PropSchema,
+  createBlockSpecFromStronglyTypedTiptapNode,
+  createStronglyTypedTiptapNode,
+} from "../../schema/index.js";
+import { defaultProps } from "../defaultProps.js";
+import { createHighlightPlugin, Parser } from "prosemirror-highlight";
+import { createParser } from "prosemirror-highlight/shiki";
+import { TextSelection } from "@tiptap/pm/state";
+import { createDefaultBlockDOMOutputSpec } from "../defaultBlockHelpers.js";
+
+type SupportedLanguageConfig = {
+  id: string;
+  name: string;
+  match: string[];
+};
+
+export const codeBlockPropSchema = {
+  ...defaultProps,
+  language: {
+    default: "",
+    values: [...bundledLanguagesInfo.map((lang) => lang.id), ""],
+  },
+} satisfies PropSchema;
+
+const CodeBlockContent = createStronglyTypedTiptapNode({
+  name: "codeBlock",
+  content: "inline*",
+  group: "blockContent",
+  marks: "",
+  code: true,
+  defining: true,
+  addOptions() {
+    return {
+      indentLineWithTab: true,
+      supportedLanguages: [
+        {
```

Comment:
I also wasn't sure about this. Usually, in HTML/MD, code blocks without language simply leave it out, so I made an empty string the default and assigned a *Code* label to it. I think the empty string makes sense, but label can be changed

Reply:
I think we should have a `text` language (https://shiki.style/languages#plain-text) for plain text.

Then we can still decide to upon creation initialize with plain text vs javascript. Maybe let's do javascript for now because otherwise people don't really notice the syntax highlighting

Reply:
Ok, I think I can add `defaultLanguage` option with `text` as the default.
For `text`, `txt`, `plain` values, I understand `renderHTML` should skip them, not to add `class="language-text"` to the HTML?